### ISSUE TITLE
`showAdaptiveDialog` is mixing Cupertino and Material

### ISSUE THREAD

**Issue by feinstein** [flutter/flutter#159114]
### Steps to reproduce

1. Run the code below
2. See how Material and Cupertino are mixing up

### Expected results

A Cupertino Dialog should be shown with Cupertino styles only

### Actual results

The material theme bleeds into the cupertino widgets:

<img width="334" alt="image" src="https://github.com/user-attachments/assets/99015403-b547-4a94-986f-af866e05007b">

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [showDialog].

void main() => runApp(const ShowDialogExampleApp());

class ShowDialogExampleApp extends StatelessWidget {
  const ShowDialogExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: DialogExample(),
    );
  }
}

class DialogExample extends StatelessWidget {
  const DialogExample({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('showDialog Sample')),
      body: Center(
        child: OutlinedButton(
          onPressed: () => _dialogBuilder(context),
          child: const Text('Open Dialog'),
        ),
      ),
    );
  }

  Future<void> _dialogBuilder(BuildContext context) {
    return showAdaptiveDialog<void>(
      context: context,
      builder: (BuildContext context) {
        print(Theme.of(context).platform);

        return AlertDialog.adaptive(
          title: const Text('Basic dialog title'),
          content: const Text(
            'A dialog is a type of modal window that\n'
            'appears in front of app content to\n'
            'provide critical information, or prompt\n'
            'for a decision to be made.',
          ),
          actions: <Widget>[
            TextButton(
              child: const Text('Disable'),
              onPressed: () {
                Navigator.of(context).pop();
              },
            ),
            TextButton(
              child: const Text('Enable'),
              onPressed: () {
                Navigator.of(context).pop();
              },
            ),
          ],
        );
      },
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

I have run this from the docs example page [here](https://api.flutter.dev/flutter/material/showDialog.html), just modifying the code a bit.

**Comment by ahyangnb** [comment]
please use CupertinoButton in the actions of AlertDialog.adaptive.

**Comment by ahyangnb** [comment]
> please use CupertinoButton in the actions of AlertDialog.adaptive.

or we can find if any Adaptive button in flutter sdk.

**Comment by feinstein** [comment]
What's the point of having an adaptive dialog, and adaptive show command, if my inner code still needs to contain logic to select what buttons I will be using?

**Comment by huycozy** [comment]
I think @ahyangnb is right, this is working as intended since you passed [TextButton](https://api.flutter.dev/flutter/material/TextButton-class.html) which is a material component into `actions`. 

If you want to have adaptive action buttons, you should build `adaptiveAction` like this: 

https://github.com/flutter/flutter/blob/57dca6946d2e314958119e22e5b9cda7339075b9/examples/api/lib/material/dialog/adaptive_alert_dialog.0.dart#L33-L49

**Comment by feinstein** [comment]
Then the adaptive dialog is incomplete and this code should be part of the framework.

**Comment by MitchellGoodwin** [comment]
This has come up before #134347.

Summarizing my comment from that thread, but an adaptive action widget wasn't created intentionally, due to the fact that you can use broadly any kind of widget for the actions, and we didn't want to limit what could be added to the adaptive version. Plus since the basic version of the adaptive action would just be a switch statement, so adding it was of low value.

We'll keep this issue open though, and I'll bring this up for discussion with others. I am going to adjust the issue title and add some to the description.

**Comment by feinstein** [comment]
Thanks for context. I agree with part of what you said... Yes, anything can be put in the dialog, but adaptive dialogs are made to be shown looking like system dialogs, as those are usually very simple and only cove few cases, so as far as I can view, we could cover those system dialogs (e.g. row of buttons with message and title)

**Comment by dkwingsmt** [comment]
Thank you for bringing up this issue and the discussion.

To start with, `AlertDialog.adaptive` does include an example in its [documentation](https://api.flutter.dev/flutter/material/AlertDialog/AlertDialog.adaptive.html) to demonstrate the importance of providing the appropriate button class. While I understand that relying on a switch case might not offer the most seamless developer experience, even if we are to introduce an `AdaptiveDialogButton`, the TextButton class itself will still not be inherently adaptive.

Adaptive widgets, as a concept, are inherently complex. Based on our past experience, there isn’t a one-size-fits-all solution, only trade-offs. One of the major challenges lies in designing an API that strikes the right balance between flexibility and simplicity while accommodating diverse design languages. Because of these challenges, we’ve been cautious about introducing new adaptive widgets in recent years, as their advantages don’t always outweigh the added complexity they bring to the API.

One might argue to make `TextButton` adapt its appearance based on context, and that approach certainly aligns with some design philosophies. However, Flutter has chosen a different path, as outlined in the [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-interleaving-multiple-concepts-together). Flutter widgets generally allow developers to specify child widgets explicitly, leaving it to the developer to select the most appropriate ones. This approach prioritizes customizability and avoids additional code complexity in implementation. For instance, in dialogs, Flutter doesn’t restrict developers to predefined buttons: they can use any widget as a dialog action, even non-buttons. While this flexibility can require more effort from the developer, it also offers greater control, especially when compared to frameworks like SwiftUI that limit customizations to a few fixed variants.

For these reasons, I believe the current behavior and API design are intentional, and at this point, I don’t think introducing an adaptive action class is the right step. That said, I completely agree that Flutter can benefit from a smoother approach to writing adaptive widgets, and it's something I hope can be explored in the future.

I’ll be closing this issue for now, but I genuinely welcome further comments or suggestions if you’d like to share more thoughts or ideas. Thank you again for your input.

**Comment by feinstein** [comment]
I completely agree with your arguments, but I still think there's some room here for a repetitive and predictable case.

Most OS, if not all, and most apps, have a Dialog with a title, subtitle, positive action button, and optional negative button. I keep making my own `showMessageDialog` and making it adaptive, but I think this is such a repetitive component, used so broadly, that Flutter could package this in a simple API.

I am not saying to change the current API, or change current widgets, but providing this kind of adaptive dialog would avoid lots of repetitive boilerplate.

I also understand this might look like it goes against some of Flutter architecture decisions, but many exceptions like this have been made when a repetitive pattern has been identified, like in: https://github.com/flutter/flutter/issues/55378

**Comment by feinstein** [comment]
Found this package https://pub.dev/packages/adaptive_dialog, although I don't like to add another package, this might be a good solution for most people.

**Comment by dkwingsmt** [comment]
Yeah, I suggest going with 3rd party libraries that provide convenient wrappers if there is one. There are many things that are not suitable included in the Flutter framework itself, but are suitable as a 3rd party library, such as APIs that are hard to make comprehensive enough on customization or stable enough.
